src/EditorFeatures/CSharpTest/SplitComment/SplitCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,260 @@
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SplitComment
+{
+    [UseExportProvider]
+    public class SplitCommentCommandHandlerTests
+    {
+        /// <summary>
+        /// verifyUndo is needed because of https://github.com/dotnet/roslyn/issues/28033
+        /// Most tests will continue to verifyUndo, but select tests will skip it due to
+        /// this known test infrastructure issure. This bug does not represent a product
+        /// failure.
+        /// </summary>
+        private void TestWorker(
+            string inputMarkup,
+            string expectedOutputMarkup,
+            Action callback,
+            bool verifyUndo = true,
+            IndentStyle indentStyle = IndentStyle.Smart)
+        {
+            using var workspace = TestWorkspace.CreateCSharp(inputMarkup);
+            var workspaceOptions = workspace.Options.WithChangedOption(new OptionKey(SmartIndent, LanguageNames.CSharp), indentStyle);
+            bool workspaceChanges = workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(workspaceOptions));
+            Assert.True(workspaceChanges);
+
+            var document = workspace.Documents.Single();
+            var view = document.GetTextView();
+
+            var originalSnapshot = view.TextBuffer.CurrentSnapshot;
+            var originalSelections = document.SelectedSpans;
+
+            var snapshotSpans = new List<SnapshotSpan>();
+            foreach (var selection in originalSelections)
+            {
+                snapshotSpans.Add(selection.ToSnapshotSpan(originalSnapshot));
+            }
+            view.SetMultiSelection(snapshotSpans);
+
+            var undoHistoryRegistry = workspace.GetService<ITextUndoHistoryRegistry>();
+            var commandHandler = new SplitCommentCommandHandler(
+                undoHistoryRegistry,
+                workspace.GetService<IEditorOperationsFactoryService>());
+
+            if (!commandHandler.ExecuteCommand(new ReturnKeyCommandArgs(view, view.TextBuffer), TestCommandExecutionContext.Create()))
+            {
+                callback();
+            }
+
+            if (expectedOutputMarkup != null)
+            {
+                MarkupTestFile.GetSpans(expectedOutputMarkup,
+                    out var expectedOutput, out ImmutableArray<TextSpan> expectedSpans);
+
+                Assert.Equal(expectedOutput, view.TextBuffer.CurrentSnapshot.AsText().ToString());
+
+                if (verifyUndo)
+                {
+                    // Ensure that after undo we go back to where we were to begin with.
+                    var history = undoHistoryRegistry.GetHistory(document.GetTextBuffer());
+                    history.Undo(count: originalSelections.Count);
+
+                    var currentSnapshot = document.GetTextBuffer().CurrentSnapshot;
+                    Assert.Equal(originalSnapshot.GetText(), currentSnapshot.GetText());
+                }
+            }
+        }
```

Comment:
i assume you have to do most of this for your vB tests.  can we not share this code?